modules/services/podman-linux/builds.nix

```
@@ -0,0 +1,173 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.podman;
+
+  podman-lib = import ./podman-lib.nix { inherit lib config; };
+
+  createQuadletSource = name: buildDef:
+    let
+      buildConfig = (podman-lib.deepMerge {
+        Build = {
+          AuthFile = buildDef.authFile;
+          Environment = buildDef.environment;
+          File = buildDef.file;
+          ImageTag = ([ "homemanager/${name}" ] ++ buildDef.tags);
+          Label = (buildDef.labels // { "nix.home-manager.managed" = true; });
+          PodmanArgs = buildDef.extraPodmanArgs;
+          SetWorkingDirectory = buildDef.workingDirectory;
+          TLSVerify = buildDef.tlsVerify;
+        };
+        Install = {
+          WantedBy = (if buildDef.autoStart then [
+            "default.target"
+            "multi-user.target"
+          ] else
+            [ ]);
+        };
+        Service = {
+          TimeoutStartSec = 300;
+          RemainAfterExit = "yes";
+        };
+        Unit = {
+          Description = (if (builtins.isString buildDef.description) then
+            buildDef.description
+          else
+            "Service for build ${name}");
+        };
+      } buildDef.extraConfig);
+    in ''
+      # Automatically generated by home-manager for podman build configuration
+      # DO NOT EDIT THIS FILE DIRECTLY
+      #
+      # ${name}.build
+      ${podman-lib.toQuadletIni buildConfig}
+    '';
+
+  toQuadletInternal = name: buildDef: {
+    assertions = podman-lib.buildConfigAsserts name buildDef.extraConfig;
+    serviceName =
+      "podman-${name}"; # quadlet service name: 'podman-<name>-build.service
+    source = podman-lib.removeBlankLines (createQuadletSource name buildDef);
+    resourceType = "build";
+  };
+in let
+  buildDefinitionType = types.submodule {
+    options = {
+
+      autoStart = mkOption {
+        type = types.bool;
+        default = true;
+        description =
+          "Whether to start the build on boot. (Requires user lingering)";
+      };
+
+      authFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = "Path of the authentication file.";
+      };
+
+      description = mkOption {
+        type = with types; nullOr str;
+        default = null;
```

Can change `types.submodule {` to `types.submodule ({name, ...}: {` and then change this like so:
```suggestion
        type = with types; nullOr str;
        default = "Service for build ${name}";
        defaultText = "Service for build \${name}";
```